hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/qjournal/client/QuorumJournalManager.java

```diff
@@ -667,6 +700,9 @@ AsyncLoggerSet getLoggerSetForTests() {
 
   @Override
   public void doPreUpgrade() throws IOException {
+    if (isEnableJnMaintenance()) {
+      throw new IOException("doPreUpgrade() does not support enabling jn maintenance mode");
```

Comment:
nit: doPreUpgrade() does not support enabling jn maintenance mode -> doPreUpgrade() is not support while in jn maintenance mode